### PR TITLE
[Feature](bangc-ops): merge three_interpolate_forward and three_inter…

### PR DESCRIPTION
**Pull request by ArtIntAI**
…polate_backward kernel code

Thanks for your contribution and we appreciate it a lot. 

## 1. Motivation

Please describe your motivation and the goal you want to achieve through this pull request.

## 2. Modification

Please briefly describe what modification is made in this pull request, and indicate where to make the modification.

## 3. Test Report

If you want to know how to do operator testing, you can see [GTest-User-Guide-zh](../docs/GTest-User-Guide-zh.md).

#### 3.1.1 Accuracy Acceptance Standard

For static threshold standard details, see: [MLU-OPS Accuracy Acceptance Standard](../docs/MLU-OPS-Accuracy-Acceptance-Standard.md).

- [x] diff1, diff1 <= 3e-3
- [x] diff2, diff2 <= 3e-3

#### 3.1.2 Operator Scheme checklist

|     No.      |           Details            |      Check Results       |
|----------------|---------------------------|---------------------|
|        1       |          Supported hardware         | MLU270 <br> MLU290 <br>MLU370<br>MLU590|
|        2       |          Job types          |    block <br> U1 <br> U4    |
|        3       |         Layouts            |  NHWC 、NCHW、ARRAY etc    |
|        4       |         Whether multi-dimensions are supported              |                |
|        5       |          Whether element zero is supported             |               |
|        6       |         Data type(half/float)       |         half / float etc           |
|        7      |        Whether there is size limit           |             |

#### 3.1.3 New Feature Test

If you have checked the following items, please tick the relevant box.

- [x] Data type test
- [ ] Multidimensional tensor test
- [ ] Layout test
- [x] Different size/integer remainder end segment/alignment misalignment test
- [x] Zero dimensional tensor test/zero element test
- [x] stability test
- [x] Multiple platform test
- [x] Gen_case module test
- [x] Nan/INF tests 
- [ ] Bug fix tests
- [x] For memory leak check details, see[GTest-User-Guide-zh](../docs/GTest-User-Guide-zh.md).
- [x] For code coverage check details, see: [GTest-User-Guide-zh](../docs/GTest-User-Guide-zh.md).
- [x] For I/O calculation efficiency check details, see: [MLU-OPS Performance Acceptance Standard](../docs/MLU-OPS-Performance-Acceptance-Standard.md).

#### 3.1.4 Parameter Check

When a new operator is submitted, the test points are given and the test results are stated.

| Test Point         | Acceptance Standard | Test Result (Error Message) |
| -------------- | -------- | -------------------- |
| Whether it conforms to the operator restriction | Normal error |                      |
| Whether illegal parameters are passed  | Normal error |                      |

### 3.2 Performance Test

See [MLU-OPS Performance Acceptance Standard](../docs/MLU-OPS-Performance-Acceptance-Standard.md) for details.

Platform ：MLU270

|Operator|Mlu_hardware_time(us)|Mlu_interface_time(us)|Mlu_io_efficiency|Mlu_compute_efficiency|Mlu_workwpace_size(Bytes)|Data_type|Shape|
|-----|----|----|----|----|----|------|-----|
|op_name|   |    |     |    |    |    |     |
|op_name|   |    |     |    |    |    |     |

Platform ：MLU290

|Operator|Mlu_hardware_time(us)|Mlu_interface_time(us)|Mlu_io_efficiency|Mlu_compute_efficiency|Mlu_workwpace_size(Bytes)|Data_type|Shape|
|-----|----|----|----|----|----|------|-----|
|op_name|   |    |     |    |    |    |     |
|op_name|   |    |     |    |    |    |     |

Platform：MLU370

|Operator|Mlu_hardware_time(us)|Mlu_interface_time(us)|Mlu_io_efficiency|Mlu_compute_efficiency|Mlu_workwpace_size(Bytes)|Data_type|Shape|
|-----|----|----|----|----|----|------|-----|
|op_name|   |    |     |    |    |    |     |
|op_name|   |    |     |    |    |    |     |

Platform：MLU590

|Operator|Mlu_hardware_time(us)|Mlu_interface_time(us)|Mlu_io_efficiency|Mlu_compute_efficiency|Mlu_workwpace_size(Bytes)|Data_type|Shape|
|-----|----|----|----|----|----|------|-----|
|op_name|   |    |     |    |    |    |     |
|op_name|   |    |     |    |    |    |     |

### 3.3 Summary Analysis

Please give a brief overview here, if you want to note and summarize the content.

### CODE REVIEW

File: bangc-ops/kernels/three_interpolate/three_interpolate.cpp [hunk 1]
```diff
@@ -34,11 +34,174 @@
...
-#define OUTPUT_MC_LIMIT_SIZE 2
+#define OUTPUT_NC_LIMIT_SIZE 2
+#define INPUT_MC_LIMIT_SIZE 2
 #define INDEX_TYPE_CONVERT_N_LIMIT_SIZE 3
 #define REPEATS_TIME_GAP 1
 #define REPEATS_TIME_MOD 0
+#define INPUT_NC_LIMIT_SIZE 2
+#define OUTPUT_MC_LIMIT_SIZE 2
+
+static void PolicyFuncThreeInterpolateForward(
```
Comment (DanieeelLiu) [inline bangc-ops/kernels/three_interpolate/three_interpolate.cpp:45]: 两个policy func 绝大部分内容相同，建议分析是否可以提出公共部分，建设冗余代码

File: bangc-ops/kernels/three_interpolate/three_interpolate_union1.mlu [hunk 2]
```diff
@@ -34,6 +34,289 @@ __nram__ char nram_buffer[MAX_NRAM_SIZE];
 #define INT32_MASK_REPEAT_TIMES 4
 #define INT16_MASK_REPEAT_TIMES 2
 
+template <typename T>
+__mlu_global__ void MLUKernelThreeInterpolateForward(
```
Comment (DanieeelLiu) [inline bangc-ops/kernels/three_interpolate/three_interpolate_union1.mlu:38]: forward backward 中kernel 代码相同部分尽量把公共逻辑部分提出来，减少代码冗余

**Review by tudejiang79 — APPROVED** [review]

File: bangc-ops/kernels/three_interpolate/three_interpolate.cpp (hunk 1 above)
Comment (ArtIntAI) [reply bangc-ops/kernels/three_interpolate/three_interpolate.cpp]: 已经修改。

File: bangc-ops/kernels/three_interpolate/three_interpolate_union1.mlu (hunk 2 above)
Comment (ArtIntAI) [reply bangc-ops/kernels/three_interpolate/three_interpolate_union1.mlu:124]: 已经修改。

**Review by DanieeelLiu — APPROVED** [review]

**Review by GuoZhiBin2014 — APPROVED** [review]

**Review by LittleReal — APPROVED** [review]
防呆修改ok

**Pull request merged**